### PR TITLE
[Breakpoints] Switch Pending BPs from Immutable to Pojo

### PR DESCRIPTION
**Pull request by jasonLaster**
issue: #6279

### Summary of Changes

* switches pending breakpoints to a plain object
* this will make it easier to persist / serialize 
* it also will help us flow type breakpoints

### Test Plan

flow and jest were clutch here

### CODE REVIEW

File: src/reducers/pending-breakpoints.js [hunk 1]
```diff
@@ -87,25 +70,23 @@ function update(
 }
 
 function addBreakpoint(state, action) {
-  if (action.status !== "done") {
+  if (action.breakpoint.hidden || action.status !== "done") {
```
Comment (wldcordeiro) [inline src/reducers/pending-breakpoints.js]: Slightly different condition? Does this fix a bug too?

Comment (jasonLaster) [reply src/reducers/pending-breakpoints.js]: cleans up the code a bit

File: src/reducers/pending-breakpoints.js [hunk 2]
```diff
@@ -21,33 +18,19 @@ import { prefs } from "../utils/prefs";
...
 
 function update(
-  state: Record<PendingBreakpointsState> = initialPendingBreakpointsState(),
+  state: PendingBreakpointsState = initialPendingBreakpointsState(),
   action: Action
 ) {
   switch (action.type) {
     case "ADD_BREAKPOINT": {
-      if (action.breakpoint.hidden) {
-        return state;
```
Comment (AnshulMalik) [inline src/reducers/pending-breakpoints.js]: This is not required?

Comment (jasonLaster) [reply src/reducers/pending-breakpoints.js]: It is moved to `addBreakpoint`. Thanks for asking

**Pull request merged**